### ISSUE TITLE
ModuleNotFoundError: No module named 'rdy2cpl.namcouple'

### ISSUE THREAD

**Issue by NishkaDG** [uwefladrich/rdy2cpl#36]
Hello,
I am trying to install and run rdy2cpl inside a virtual environment created by building the wheels and installing them with Python venv (not conda or mamba). 
Python version: 3.9.13
OS: RHEL 8.6 Ootpa
I get the following error when I run scriptengine `se --help`:
`$ se --help
Traceback (most recent call last):
  File "/home/user/.venv/bin/se", line 8, in <module>
    sys.exit(main())
  File "/home/user/.venv/lib64/python3.9/site-packages/scriptengine/cli/se.py", line 105, in main
    parsed_args = parse_cmd_line_args()
  File "/home/user/.venv/lib64/python3.9/site-packages/scriptengine/cli/se.py", line 48, in parse_cmd_line_args
    epilog="Available ScriptEngine tasks: " + ", ".join(load_tasks().keys()),
  File "/home/user/.venv/lib64/python3.9/site-packages/scriptengine/tasks/core/loader.py", line 26, in load
    loaded_tasks[ep.name] = ep.load()
  File "/usr/lib64/python3.9/importlib/metadata.py", line 77, in load
    module = import_module(match.group('module'))
  File "/usr/lib64/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/home/user/.venv/lib64/python3.9/site-packages/rdy2cpl/scriptengine_task.py", line 10, in <module>
    from rdy2cpl.namcouple.factory import from_yaml
ModuleNotFoundError: No module named 'rdy2cpl.namcouple'`
I tried to use the advice in #19 (creating blank \_\_init\_\_.py files in directories rdy2cpl and rdy2cpl/namcouple) but that does not seem to have worked.
Is there anything I can do to get it to run?
I am also getting an error on `r2c --help`.
Thanks!

**Comment by uwefladrich** [comment]
Hi,

I am not entirely sure where it is going wrong, but I made another test to install in a virtualenv. However, the normal mode of operation is with conda/mamba env, so virtualenvs are less tested, I'm afraid.

Here is the test log:

```bash
> git clone https://github.com/uwefladrich/rdy2cpl.git
> cd rdy2cpl/

> python --version
Python 3.9.18

> python -m venv .venv
> source .venv/bin/activate

> pip install -U pip
# [...]
> pip install scriptengine
# [...]
> pip install mpi4py eccodes  # not sure this is actually necessary or pulled by the next step anyway
# [...]
> pip install -U -e .
# [...]
> se --help
usage: se [-h] [-V] [--loglevel {debug,info,warning,error,critical}] [--nocolor] files [files ...]
# [...]
Available ScriptEngine tasks: base.chdir, base.command, base.context, base.context.from, base.copy, base.echo, base.exit, base.find, base.getenv, base.include, base.link, base.make_dir, base.move,
base.remove, base.setenv, base.task_timer, base.template, base.time, chdir, command, context, context.from, copy, echo, exit, find, getenv, include, link, make_dir, move, remove, setenv, task_timer,
template, time, rdy2cpl.make_all, rdy2cpl.make_namcouple
```

Could you try that with a fresh checkout?

**Comment by NishkaDG** [comment]
Thank you. 
Unfortunately we have to build this without pip/conda/mamba on Boreas, so we were trying to reinvent the wheel by using pip wheel to build everything. It seems that the process finished but without the MPI bindings, so se --help threw that error on namcouple.

**Comment by jbrnlz** [comment]
Hi both,

1) When building wheels locally (aiming at rsync them later to a no-internet machine like boreas), the trick is to let pip do it from the repos directly instead of cloning them locally. Something like the following one-liner:

`python -m pip wheel git+https://github.com/ecmwf/eccodes-python git+https://github.com/uwefladrich/scriptengine.git@master git+https://github.com/uwefladrich/scriptengine-tasks-hpc@master git+https://github.com/uwefladrich/rdy2cpl.git@main`

The resulting set of wheels will then be dep-consistent, and when building them on your target machine, your `se --help` call will exit with no errors and including all tasks Uwe shows above.

2) The MPI issue is caused by Python not being able to find the `libmpi.so.40` library, which can be solved by loading the `openmpi-4.1.4*` module on boreas.

3) The rest of the errors (those you see when calling `r2c --help`) are gone after you build OASIS using SE and [connecting them via env variables](https://ec-earth-4-docs.readthedocs.io/en/latest/building/main.html#completing-the-environment), e.g.:

`export PYOASIS_ROOT=/path/to/ece4/oasis3-mct-5.2/arch_ecearth
export LD_LIBRARY_PATH=${PYOASIS_ROOT}/lib:${LD_LIBRARY_PATH}
export PYTHONPATH=${PYOASIS_ROOT}/python:${PYTHONPATH}
export OASIS_BUILD_PATH=$PYOASIS_ROOT`

After that both `se --help` and `r2c --help` should exit as expected.
